shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -124,7 +125,29 @@ - (void)userNotificationCenter:(UNUserNotificationCenter*)center
 - (BOOL)application:(UIApplication*)application
             openURL:(NSURL*)url
             options:(NSDictionary<UIApplicationOpenURLOptionsKey, id>*)options {
-  return [_lifeCycleDelegate application:application openURL:url options:options];
+  if ([_lifeCycleDelegate application:application openURL:url options:options]) {
+    return YES;
+  } else {
+    UIViewController* rootViewController = _window.rootViewController;
+    if ([rootViewController isKindOfClass:[FlutterViewController class]]) {
+      FlutterViewController* flutterViewController = (FlutterViewController*)rootViewController;
+      [flutterViewController.engine
+          waitForFirstFrame:3.0
+                   callback:^(BOOL didTimeout) {
+                     if (didTimeout) {
+                       FML_LOG(ERROR)
+                           << "Timeout waiting for the first frame when launching an URL.";
+                     } else {
+                       [flutterViewController.engine.navigationChannel invokeMethod:@"pushRoute"
+                                                                          arguments:url.path];
+                     }
+                   }];
+      return YES;
+    } else {
+      FML_LOG(ERROR) << "Attempting to open an URL without a Flutter RootViewController.";
```

Comment:
In what situation this can happen?

Reply:
It's unlikely but if you edit the storyboard for the iOS project this could happen.  Also, if you were doing add to app and you call into a FlutterAppDelegate it could be a problem.